### PR TITLE
clean and refactor elastic serach ETL job

### PR DESCRIPTION
**Pull request by camelliazhang**
In this commit, I have 
1, Add index_mapping json file
2, Modify index_mappings.md file to include the instructions such as building index, create index alias, etc
3, Modify the data set update SQL using left Join
4, Make default value for static boosting score as 0, to avoid null data populating issues
5, Rewrite code for flow_jobs, metric, etc, to make it more readable. 
6, Add a parameter for elastic index version, to make it configurable through wh_propery

### CODE REVIEW

File: metadata-etl/src/main/resources/jython/ElasticSearchIndex.py [hunk 1]
```diff
@@ -27,6 +27,8 @@ def __init__(self, args):
     self.logger = LoggerFactory.getLogger('jython script : ' + self.__class__.__name__)
     self.elasticsearch_index_url = args[Constant.WH_ELASTICSEARCH_URL_KEY]
     self.elasticsearch_port = args[Constant.WH_ELASTICSEARCH_PORT_KEY]
+    self.elasticsearch_index = args[Constant.WH_ELASTICSEARCH_INDEX_KEY]
```
Comment (ericsun2) [inline metadata-etl/src/main/resources/jython/ElasticSearchIndex.py]: Do we set any default value for this param?

Comment (camelliazhang) [inline metadata-etl/src/main/resources/jython/ElasticSearchIndex.py]: Not anywhere in the code. or should we set it as "wherehows" if nothing  in args[Constant.WH_ELASTICSEARCH_INDEX_KEY] is found?
Looks like We do not have such checks on other parameters yet

File: data-model/ELASTICSEARCH/index_mappings.md [hunk 2]
```diff
@@ -1,43 +1,44 @@
...
-        "type": "dataset"
-      }
-    }
-  }
-}
-'
 
-curl -XPUT '$YOUR_INDEX_URL:9200/wherehows/flow_jobs/_mapping' -d '
+https://www.elastic.co/guide/index.html
+
```
Comment (ericsun2) [inline data-model/ELASTICSEARCH/index_mappings.md]: Please explain the intention and convention of using alias "wherehows" here.

The create step index step needs to reflect the usage of alias.

File: metadata-etl/src/main/resources/jython/ElasticSearchIndex.py (hunk 1 above)
Comment (camelliazhang) [inline metadata-etl/src/main/resources/jython/ElasticSearchIndex.py]: Just added the default index as "wherehows", and tested

File: data-model/ELASTICSEARCH/index_mappings.md (hunk 2 above)
Comment (camelliazhang) [inline data-model/ELASTICSEARCH/index_mappings.md]: More description added, could you please take another look? Thanks! Na

**Pull request merged**